### DISCUSSION TITLE
Cosmos DB retry policy

### DISCUSSION THREAD

**Discussion by tobbin2** [Azure/azure-cosmos-dotnet-v3#4092] [Q&A] [answered]
Hello,
this might be a dumb question but i just want it verified. Due to the documentation does it say that this SDK should retry automatically in case of some responses such as 429 request limiter.

If it now already retries in the service/server level, do i still receive an 429 response in the request? If i don't is it still possible to get one in case that the retries already was performed by the sevice/server but it still fails?

**Comment by bartelink** [comment]
you set a max retries count and time wrt rate limiting. When that's exhausted, the exception will be thrown in the normal way.

>If i don't is it still possible to get one in case that the retries already was performed by the sevice/server but it still fails?

there is no service/server factoe
rate limiting retries are all driven from the client side

Other retries are your business

**Reply by tobbin2** [reply]
okay i understand.
So if i don't set any max retries or time wrt rate limiting, do i have to handle it manually or does it still retry until success.
I understand that its handled by client side, however when i execute the request to CosmosDb can i catch an CosmosDBException, so in case the "retry count" already retried, will i still get that exception or is it handled as a success.

**Reply by bartelink** [reply, marked as answer]
default limits are documented. something like 9 tries and 30s. There can always be an exception. so if you config max 3 and 15s, you can get thrown an exn if attempt 4 fails or 15s has expired
if you dont get an exception, the request got servied

**Reply by tobbin2** [reply]
Okay thanks for the quick response. It was really helpful 👌
